repository/managers.py

```diff
@@ -211,12 +222,18 @@ def item(self, groupId, itemId, fileId):
             A dictionary containing item details and Giles document text.
             
         Raises:
-            CitesphereAPIError
+            GilesUploadError: When there's an issue with Giles uploads
+            GilesTextExtractionError: When text content cannot be extracted
         """
         # Fetch item details using CitesphereAPIv1
-        item_data = self.api.get_item_details(groupId, itemId)
+        try:
+            item_data = self.api.get_item_details(groupId, itemId)
+        except CitesphereAPIError as e:
+            logger.error(f"Failed to fetch item details: {str(e)}")
+            raise
```

Comment:
exceptions should be handled where they are caught. So no logging here if you just pass on the exception (in fac you can remove the whole try except block)